### PR TITLE
Jpn translit

### PR DESCRIPTION
**Pull request by MadWasp79**
For review purposes only. Do not accept.

### CODE REVIEW

File: OsmAnd/src/net/osmand/AndroidUtils.java [hunk 1]
```diff
@@ -524,4 +524,5 @@ public int compare(Map.Entry<K, V> o1, Map.Entry<K, V> o2) {
 		}
 		return result;
 	}
+
```
Comment (crimean) [inline OsmAnd/src/net/osmand/AndroidUtils.java]: ?

File: OsmAnd/src/net/osmand/plus/render/TextRenderer.java [hunk 2]
```diff
@@ -29,9 +31,12 @@
 import android.graphics.Rect;
 import android.graphics.RectF;
 import android.graphics.Typeface;
+import org.apache.commons.logging.Log;
 
 public class TextRenderer {
 
+	public final static Log LOG = PlatformUtil.getLog(TextRenderer.class);
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/render/TextRenderer.java]: ?

File: OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+public class TransliterationHelper {
+
+	private static TransliterationHelper instance;
+
+	public final static Log LOG = PlatformUtil.getLog(TransliterationHelper.class);
+	public final static String DEFAULT = "default";
+	public final static String JAPAN = "Japan";
+
+	private static String country = DEFAULT;
```
Comment (crimean) [inline OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java]: countryName?

File: OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+			}
+		}
+	}
+
+	public static String getCountry() {
+		return country;
+	}
+
+	public static String transliterate(String text) {
+		if (tokenizer == null) {
```
Comment (crimean) [inline OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java]: do init tokenizer in japanese2Romaji

File: OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+
+	static {
+		try {
+			instance = new TransliterationHelper();
+		} catch (Exception e) {
+			LOG.debug(e.getMessage(), e);
+		}
+	}
+
+	public static TransliterationHelper getInstance() {
```
Comment (crimean) [inline OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java]: is it needed?

File: OsmAnd/src/net/osmand/plus/render/MapRenderRepositories.java [hunk 6]
```diff
@@ -303,8 +306,11 @@ public void checkInitialized(final int zoom, NativeOsmandLibrary library, int le
 		boolean useLive = context.getSettings().USE_OSM_LIVE_FOR_ROUTING.get();
 		for (String mapName : files.keySet()) {
 			BinaryMapIndexReader fr = files.get(mapName);
-			if (fr != null && (fr.containsMapData(leftX, topY, rightX, bottomY, zoom) || 
+			if (fr != null && (fr.containsMapData(leftX, topY, rightX, bottomY, zoom) ||
 					fr.containsRouteData(leftX, topY, rightX, bottomY, zoom))) {
+				if (!Algorithms.isEmpty(fr.getCountryName())) {
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/render/MapRenderRepositories.java]: what if few files will contain map data and first one is japan but last one is not japan?

File: OsmAnd/src/net/osmand/plus/render/TextRenderer.java (hunk 2 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/render/TextRenderer.java]: changed to private

File: OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java (hunk 3 above)
Comment (MadWasp79) [reply OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java]: +

File: OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java (hunk 4 above)
Comment (MadWasp79) [reply OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java]: +

File: OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java (hunk 5 above)
Comment (MadWasp79) [reply OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java]: nope

File: OsmAnd/src/net/osmand/plus/render/MapRenderRepositories.java (hunk 6 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/render/MapRenderRepositories.java]: fixed

File: OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+
+import com.atilika.kuromoji.ipadic.Token;
+import com.atilika.kuromoji.ipadic.Tokenizer;
+import net.osmand.PlatformUtil;
+import net.sf.junidecode.Junidecode;
+import org.apache.commons.logging.Log;
+
+public class TransliterationHelper {
+
+	private static TransliterationHelper instance;
```
Comment (crimean) [inline OsmAnd-java/src/main/java/net/osmand/util/TransliterationHelper.java]: ?

File: OsmAnd/build.gradle [hunk 8]
```diff
@@ -66,6 +66,12 @@ android {
 		warningsAsErrors false
 	}
 
+	packagingOptions {
```
Comment (crimean) [inline OsmAnd/build.gradle]: ?

File: OsmAnd/build.gradle [hunk 9]
```diff
@@ -391,7 +397,7 @@ dependencies {
 	implementation 'com.moparisthebest:junidecode:0.1.1'
 	implementation 'org.immutables:gson:2.5.0'
 	implementation 'com.vividsolutions:jts-core:1.14.0'
-
+	implementation 'com.atilika.kuromoji:kuromoji-ipadic:0.9.0'
```
Comment (crimean) [inline OsmAnd/build.gradle]: if this library included in OsmAnd-java - do not include in OsmAnd

File: OsmAnd/src/net/osmand/plus/render/MapRenderRepositories.java [hunk 10]
```diff
@@ -1,6 +1,8 @@
 package net.osmand.plus.render;
 
 
+import static net.osmand.util.TransliterationHelper.setCountryName;
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/render/MapRenderRepositories.java]: do not import static method

**Pull request merged**